www/js/diary/list.js

```diff
@@ -277,14 +277,34 @@ angular.module('emission.main.diary.list',['ui-leaflet',
             DiaryHelper.directiveForTrip);
           Timeline.setTripWrappers(currDayTripWrappers);
 
+          let tripFromNotification = null;
+
           $scope.data.currDayTripWrappers.forEach(function(tripgj, index, array) {
             $scope.populateModeFromTimeline(tripgj, $scope.data.unifiedConfirmsResults.modes);
             $scope.populatePurposeFromTimeline(tripgj, $scope.data.unifiedConfirmsResults.purposes);
             $scope.populateSurveyAnswerFromTimeline(tripgj, $scope.data.unifiedConfirmsResults.surveyAnswers);
             $scope.populateBasicClasses(tripgj);
             $scope.populateCommonInfo(tripgj);
+
+            if($rootScope.displayingIncident == true && $rootScope.notificationData) {
+              console.log('tripgj =>', tripgj);
+              console.log('$rootScope.notificationData =>', $rootScope.notificationData);
+              if (
+                tripgj.data.properties.start_ts === $rootScope.notificationData.start_ts &&
+                tripgj.data.properties.end_ts === $rootScope.notificationData.end_ts
+              ) {
+                tripFromNotification = tripgj;
+              }
+            }
           });
           $ionicScrollDelegate.scrollTop(true);
+
+          if (tripFromNotification) {
```

Comment:
It looks like that I have to display the confirm survey here because I have to wait for `Timeline.UPDATE_DONE` instead of displaying it right after `readAndUpdateForDay`.

However, as the trips are displayed correctly on the date the notification is selected but I cannot match any trip to the notification at all. Not sure what happens.